cfb-mode/src/decrypt.rs

```diff
@@ -21,13 +21,64 @@ where
     iv: Block<C>,
 }
 
+/// CFB mode buffered decryptor.
+#[derive(Clone)]
+pub struct BufferDecryptor<C>
+where
+    C: BlockEncryptMut + BlockCipher,
+{
+    cipher: C,
+    iv: Block<C>,
+    pos: usize,
+}
+
+impl<C> BufferDecryptor<C>
+where
+    C: BlockEncryptMut + BlockCipher,
+{
+    /// Decrypt a buffer in multiple parts.
+    pub fn decrypt(&mut self, mut data: &mut [u8]) {
+        let bs = C::BlockSize::to_usize();
+        let n = data.len();
+
+        if n < bs - self.pos {
+            xor_set2(data, &mut self.iv[self.pos..self.pos + n]);
+            self.pos += n;
+            return;
+        }
+        let (left, right) = { data }.split_at_mut(bs - self.pos);
+        data = right;
+        let mut iv = self.iv.clone();
+        xor_set2(left, &mut iv[self.pos..]);
+        self.cipher.encrypt_block_mut(&mut iv);
+
+        let mut chunks = data.chunks_exact_mut(bs);
+        for chunk in &mut chunks {
+            xor_set2(chunk, iv.as_mut_slice());
+            self.cipher.encrypt_block_mut(&mut iv);
+        }
+
+        let rem = chunks.into_remainder();
+        xor_set2(rem, iv.as_mut_slice());
+        self.pos = rem.len();
+        self.iv = iv;
+    }
+}
+
 impl<C> BlockSizeUser for Decryptor<C>
 where
     C: BlockEncryptMut + BlockCipher,
 {
     type BlockSize = C::BlockSize;
 }
 
+impl<C> BlockSizeUser for BufferDecryptor<C>
+where
+    C: BlockEncryptMut + BlockCipher,
```

Comment:
I do not think we need to implement `BlockSizeUser` for buffered types.

Reply:
removed